src/GongSolutions.WPF.DragDrop/DefaultDropHandler.cs

```diff
@@ -234,6 +252,30 @@ public virtual void Drop(IDropInfo dropInfo)
             }
         }
 
+        private static void Move(IList list, int sourceIndex, int destinationIndex)
+        {
+            if (!IsObservableCollection(list))
+                throw new ArgumentException("ObservableCollection<T> was expected",nameof(list));
+            var method = list.GetType().GetMethod("Move",
+                System.Reflection.BindingFlags.Instance
+                | System.Reflection.BindingFlags.Public);            
+            //method = method.MakeGenericMethod(listItemType);
+            method.Invoke(list, new object[] { sourceIndex, destinationIndex });            
+
+        }
+
+        private static bool SameObservableCollection(IList collection1, IList collection2)
+        {
+            return Equals(collection1, collection2) && IsObservableCollection(collection1);
```

Comment:
Should use `ReferenceEquals` in this case.
```suggestion
            return ReferenceEquals(collection1, collection2) && IsObservableCollection(collection1);
```